packages/mui-base/src/SelectUnstyled/SelectUnstyled.types.ts

```diff
@@ -81,10 +86,23 @@ export interface SelectUnstyledOwnProps<TValue extends {}> extends SelectUnstyle
    * The default selected value. Use when the component is not controlled.
    */
   defaultValue?: TValue | null;
+  /**
+   * A function to convert the currently selected value to a string.
+   * Used to set a value of a hidden input associated with the select,
+   * so that the selected value can be posted with a form.
+   */
+  formValueProvider?: (option: SelectOption<TValue> | null) => string | number;
```

Comment:
Why is this here `string | number`, but in the MultiSelect it is only `string | string[]`. Should this be `string` only?

Reply:
Not sure on the name. Should it be simply `getValue`?

Reply:
> Why is this here string | number, but in the MultiSelect it is only string | string[]. Should this be string only?

The native `select`'s (and hidden `input`'s as well) value is typed as `string | number | readonly string[] | undefined`. We don't have to artificially restrict numbers.

> Should it be simply getValue?

`getValue` doesn't indicate it's a value serialized for a form. Perhaps `serializeValue` (or `getSerializedValue`, or `getFormValue`)?

Reply:
`getSerializedValue` sounds great 👍 

> The native select's (and hidden input's as well) value is typed as string | number | readonly string[] | undefined. We don't have to artificially restrict numbers.

Can we just use this type then? We can omit some values in the single select

Reply:
Good point. I suppose we don't even need to omit values in a single select - if someone really wants to convert a single value to string[], there are no technical reasons why they shouldn't.